tests/posix/common/src/pthread.c

```diff
@@ -465,6 +465,40 @@ ZTEST(pthread, test_pthread_cleanup)
 	zassert_ok(pthread_join(th, NULL));
 }
 
+static bool testcancel_failed;
+
+static void *test_pthread_cancel_fn(void *arg)
+{
+	zassert_ok(pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL));
+
+	/* this should be ignored */
+	pthread_testcancel();
```

Comment:
this call isn't supposed to terminate the thread, but if it does, the test would pass instead of fail, since `testcancel_failed` is `false` at this point

Reply:
```
static bool testcancel_failed = true;

static void *test_pthread_cancel_fn(void *arg)
{
	zassert_ok(pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL));

	/* this should be ignored */
	pthread_testcancel();

	/* this will mark it pending */
	pthread_cancel(pthread_self());

	/* enable the thread to be cancelled */
	zassert_ok(pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL));

	testcancel_failed = false;

	/* this should terminate the thread */
	pthread_testcancel();

	testcancel_failed = true;

	return NULL;
}

```

Should be okay 🤔

Reply:
The above works for testing that the last testcancel call works as expected.

Maybe use a separate static bool to verify the previous testcancel call works as expected.

Reply:
`zassert_ok(pthread_cancel(..))`